protocol/v2/ssv/validator/validator.go

```diff
@@ -62,6 +62,11 @@ func NewValidator(pctx context.Context, cancel func(), options Options) *Validat
 		dutyIDs:     hashmap.New[spectypes.BeaconRole, string](),
 	}
 
+	// Changing specssv.BeaconNode requires modifications in spec.
+	if beaconStatusChecker, ok := options.Beacon.(NodeStatusChecker); ok {
```

Comment:
if this returns false v.Beacon stays nil, we should somehow mitigate otherwise if there's no option for this to happen this check is redundant.